lib/sidekiq.rb

```diff
@@ -149,7 +149,11 @@ def self.default_server_middleware
   end
 
   def self.default_worker_options=(hash)
-    @default_worker_options = default_worker_options.merge(hash.stringify_keys)
+    # stringify
+    hash.keys.each do |key|
+      hash[key.to_s] = hash.delete(key)
+    end
+    @default_worker_options = default_worker_options.merge(hash)
```

Comment:
How about starting with a copy of the defaults, then updating it with the provided hash like this?

```ruby
@default_worker_options = default_worker_options.dup
hash.each do |key, value|
  @default_worker_options[key.to_s] = value
end
```